tests/store/model_registry/test_sqlalchemy_store.py

```diff
@@ -382,98 +382,139 @@ def search_versions(filter_string):
-            search_registered_model("source_path = 'A/D'")
+            self.search_registered_models("source_path = 'A/D'")
         assert exception_context.exception.error_code == ErrorCode.Name(INVALID_PARAMETER_VALUE)
 
         # cannot search by other params
         with self.assertRaises(MlflowException) as exception_context:
-            search_registered_model("evilhax = true")
+            self.search_registered_models("evilhax = true")
         assert exception_context.exception.error_code == ErrorCode.Name(INVALID_PARAMETER_VALUE)
 
-        # delete rm6. search should not return RM 6
-        self.store.delete_registered_model(name=name6)
+        # delete rm6. search should not return RM 5
+        self.store.delete_registered_model(name=names[-1])
 
         # equality search using name should return no names
-        self.assertEqual(set(search_registered_model("name='%s'" % name6)), set([]))
+        self.assertEqual(set(self.get_rm_names(self.search_registered_models(
+            f"name='{names[-1]}'"))), set([]))
 
         # case-sensitive prefix search using LIKE should return all the RMs
-        self.assertEqual(set(search_registered_model(f"name LIKE '{prefix}%'")),
-                         set([name1, name2, name3, name4, name5]))
+        self.assertEqual(set(self.get_rm_names(self.search_registered_models(
+            f"name LIKE '{prefix}%'"))), set(names[0:5]))
 
         # case-insensitive prefix search using ILIKE should return both rm5 and rm6
-        self.assertEqual(set(search_registered_model(f"name ILIKE '{prefix + 'RM4A'}%'")),
-                         set([name5]))
+        self.assertEqual(set(self.get_rm_names(self.search_registered_models(
+            f"name ILIKE '{prefix + 'RM4A'}%'"))), {names[4]})
+
+    def test_search_registered_model_pagination(self):
+        rms = [self._rm_maker(f"RM{i:03}").name for i in range(50)]
+        # test that pagination will return all valid results in sorted order
+        # by name ascending
+        result = self.search_registered_models("name LIKE 'RM%'", max_results=5)
+        self.assertNotEqual(result.token, None)
+        self.assertEqual(self.get_rm_names(result), rms[0:5])
+
+        result = self.search_registered_models(
+            "name LIKE 'RM%'", page_token=result.token, max_results=10)
+        self.assertNotEqual(result.token, None)
+        self.assertEqual(self.get_rm_names(result), rms[5:15])
+
+        result = self.search_registered_models(
+            "name LIKE 'RM%'", page_token=result.token, max_results=20)
+        self.assertNotEqual(result.token, None)
+        self.assertEqual(self.get_rm_names(result), rms[15:35])
+
+        result = self.search_registered_models(
+            "name LIKE 'RM%'", page_token=result.token, max_results=100)
+        # assert that page token is None
+        self.assertEqual(result.token, None)
+        self.assertEqual(self.get_rm_names(result), rms[35:])
+
```

Comment:
these tests look good -- i would write an additional test that starts with a `None` token; and repeated calls search registered models with say `max_results = 5` until no more. In each round, collect the results and ensure that backend returns all 50 RMs